### ISSUE TITLE
[Bug] Image sampling artifacts when not rendering in physical pixels

### ISSUE THREAD

**Issue by aevyrie** [mvlabat/bevy_egui#89]
How to reproduce:
- run the supplied `ui` example
- toggle scale with the `/` key

1:1 physical pixel scale:
![image](https://user-images.githubusercontent.com/2632925/162816687-74b0cfa2-6090-4f95-91f2-312dd3441c90.png)

Default DPI-aware scaling:
![image](https://user-images.githubusercontent.com/2632925/162816713-344b3a7a-36d1-4831-8f22-2a12bfe1e896.png)

This issue is specifically for the aliasing seen in the middle of the image, e.g. the bird's chest. There is a different alpha bug that causes aliasing at the border of the circle that applies to any image with transparency.

**Comment by mvlabat** [comment]
Thanks for filing the issue. I'm aware of the bug but I don't really know how to fix it. I'm quite a noob in rendering, so I'd appreciate any help (either a PR or some hints on how to fix it).

**Comment by aevyrie** [comment]
I'm not an expert either. 🙂 We might need to poke some people in the bevy/egui communities. My best guess is it has something to do with how the texture is scaled/sampled to the final render target.

**Comment by aevyrie** [comment]
@mvlabat I found the issue. It appears the default `SamplerDescriptor` uses `Nearest` instead of `Linear`. I was able to test a fix by adding the following to the `ui` example in this repo:

```rs
    if let Some(sampler) = &mut assets
        .get_mut(&images.bevy_icon)
        .map(|s| &mut s.sampler_descriptor)
    {
        sampler.mag_filter = FilterMode::Linear;
        sampler.min_filter = FilterMode::Linear;
    }
```

Linear:
![image](https://user-images.githubusercontent.com/2632925/163074379-6b4b4171-4714-4647-adf1-f5352e3c5f9e.png)

Nearest:
![image](https://user-images.githubusercontent.com/2632925/163074414-d8f0dcc7-08f0-42d6-91ec-ffedb2897d18.png)

Not quite sure what the best solution path is. Maybe the default filtering mode in bevy - where this originates from - should be Linear?
